taichi/rhi/amdgpu/amdgpu_device.cpp

```diff
@@ -56,17 +56,15 @@ DeviceAllocation AmdgpuDevice::allocate_memory_runtime(
   info.size = taichi::iroundup(params.size, taichi_page_size);
   if (params.host_read || params.host_write) {
     TI_NOT_IMPLEMENTED
-  } else if (params.use_cached) {
+  } else {
     info.ptr =
         DeviceMemoryPool::get_instance().allocate_with_cache(this, params);
     TI_ASSERT(info.ptr != nullptr);
 
     AMDGPUDriver::get_instance().memset((void *)info.ptr, 0, info.size);
-  } else {
-    info.ptr = allocate_llvm_runtime_memory_jit(params);
   }
   info.is_imported = false;
-  info.use_cached = params.use_cached;
+  info.use_cached = true;
```

Comment:
OOC: do we still need `use_cached` since it's always true now?

Reply:
Yup, `AllocInfo::use_cached` is also used in memory release to identify where the memory is allocated from: either `CachingAllocator`, `CudaMemoryPool` or `Preallocated Memory`, so as to determine the correspondent release method.